src/main/groovy/org/shipkit/internal/notes/contributors/FetcherCallable.java

```diff
@@ -0,0 +1,44 @@
+package org.shipkit.internal.notes.contributors;
+
+import org.shipkit.internal.notes.util.Function;
+
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.Callable;
+
+/**
+ * Basic function which applies values from a {@link List} of {@link V} to a given {@link Function}.
+ */
+class FetcherCallable<V, R> implements Callable<Set<R>> {
```

Comment:
This class is neatly abstracted out and should very easy to unit test, too. Nice!

Since it contains conditional logic and some control statements like looping, can you write a unit test for it?